logstash-core/lib/logstash/webserver.rb

```diff
@@ -20,7 +20,8 @@ def initialize(logger, options={})
       http_port    = options[:http_port] || 9600
       @options     = {}
       @cli_options = options.merge({ :rackup => ::File.join(::File.dirname(__FILE__), "api", "init.ru"),
-                                     :binds => ["tcp://#{http_host}:#{http_port}"] })
+                                     :binds => ["tcp://#{http_host}:#{http_port}"],
+                                     :debug => logger.debug? })
```

Comment:
and to provide more context, no logger is not passed to any webserver here. This works just as interface to puma. Logger is also used inside the sinatra app, but this is another moving component.